localstack/utils/aws/aws_stack.py

```diff
@@ -155,36 +151,3 @@ def extract_access_key_id_from_auth_header(headers: Dict[str, str]) -> Optional[
         access_id = auth.removeprefix("AWS ").split(":")
         if len(access_id):
             return access_id[0]
-
-
-# TODO remove the `internal` arg
-def mock_aws_request_headers(
-    service: str, aws_access_key_id: str, region_name: str, internal: bool = False
-) -> Dict[str, str]:
-    """
-    Returns a mock set of headers that resemble SigV4 signing method.
-    """
-    ctype = APPLICATION_AMZ_JSON_1_0
-    if service == "kinesis":
-        ctype = APPLICATION_AMZ_JSON_1_1
-    elif service in ["sns", "sqs", "sts", "cloudformation"]:
-        ctype = APPLICATION_X_WWW_FORM_URLENCODED
-
-    # For S3 presigned URLs, we require that the client and server use the same
-    # access key ID to sign requests. So try to use the access key ID for the
-    # current request if available
-    headers = {
-        "Content-Type": ctype,
-        "Accept-Encoding": "identity",
-        "X-Amz-Date": "20160623T103251Z",  # TODO: Use current date
-        "Authorization": (
-            "AWS4-HMAC-SHA256 "
-            + f"Credential={aws_access_key_id}/20160623/{region_name}/{service}/aws4_request, "
-            + "SignedHeaders=content-type;host;x-amz-date;x-amz-target, Signature=1234"
-        ),
-    }
-    if internal:
-        # TODO: This method of detecting internal calls is no longer valid
-        # We now use the `INTERNAL_REQUEST_PARAMS_HEADER` header which is set to the DTO
-        headers[HEADER_LOCALSTACK_ACCOUNT_ID] = get_aws_account_id()
-    return headers
```

Comment:
This function is moved to a new module `localstack.utils.aws.headers`. This is to avoid a circular import but also to improve code organisation